client/src/app/components/request-list/request-list.component.ts

```diff
@@ -63,6 +66,23 @@ export class RequestListComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    this.requestService.getAllDeals().subscribe((requests: ListItem[]) => this.dataSource = requests)
+    this.requestService.getAllDeals()
+      .subscribe((requests: any) => this.requestService.data.next(requests))
+  }
+
+  ngAfterContentInit() {
+    this.requestService.data
+      .subscribe((val: any) => this.dataSource = val)
```

Comment:
Я думаю без **any** будет корректней:
```suggestion
      .subscribe((val: ListItem[]) => this.dataSource = val)
```

Reply:
исправил